### ISSUE TITLE
Too many classes on the bootclasspath

### ISSUE THREAD

**Issue by wmdietl** [typetools/checker-framework#700]
[This issue is motivated by a discussion here: https://github.com/typetools/annotation-tools/commit/6ded22f83487938efe4709d95c807e8288ed86d1]

Take this code:

```
import com.sun.tools.javac.Main;
import org.checkerframework.checker.nullness.NullnessChecker;
class XXX {}
```

It produces two errors with standard javac (with nothing on your classpath):

```
XXX.java:1: error: package com.sun.tools.javac does not exist
import com.sun.tools.javac.Main;
                          ^
XXX.java:2: error: package org.checkerframework.checker.nullness does not exist
import org.checkerframework.checker.nullness.NullnessChecker;
                                            ^
2 errors
```

That is, the compiler `com.sun.tools.javac.Main` doesn't put itself on the bootclasspath when compiling application code.

When using the Checker Framework javac, this example compiles without errors, because javac, the Checker Framework, and many other classes are put on the bootclasspath.

Using checker/bin/javac should give the same errors as normal javac.

This should not impact the use of `org.checkerframework.checker.nullness.NullnessChecker` as a `javac -processor` argument. That is, the processorpath should include the Checker Framework.

The only point for discussion is whether the qualifiers should be automatically included or not.
That is, when using the Checker Framework javac without extra flags, should

```
import org.checkerframework.checker.nullness.qual.Nullable;
class XXX {}
```

compile with or without errors?

Raising an error would remind the user that the resulting classfile references the annotation (though no harm occurs if the annotation is not available at run time: see http://stackoverflow.com/questions/3567413/why-doesnt-a-missing-annotation-cause-a-classnotfoundexception-at-runtime).
If they have to put the checker-qual.jar file on the classpath for compilation, that will be more obvious.

Not raising an error would make it easier to use the examples, but obscure the dependency.

It might be useful to split checker-qual.jar into two files: one that only contains the annotations and another that contains the runtime utility classes.
Depending only on the annotations will usually not cause the classfile to fail loading.
We could then implicitly include the qualifier-only jar file, but require explicit inclusion of the runtime utility classes.

**Comment by wmdietl** [comment]
This is relevant to the discussion we had about annotations being a dependency:
http://stackoverflow.com/questions/3567413/why-doesnt-a-missing-annotation-cause-a-classnotfoundexception-at-runtime

**Comment by wmdietl** [comment]
Now that this is merged, should we remove afuplume.jar from annotation-tools?

**Comment by mernst** [comment]
Yes, we should.  I was afraid to do it so close to the release and was going to get to it right afterward.  (It shouldn't be hard if someone else wants to, but there is always a risk of spending more time than anticipated.)

**Comment by smillst** [comment]
Can this issue and #711 be closed?
